src/main/java/nl/tudelft/lifetiles/graph/models/jgrapht/JGraphTGraphAdapter.java

```diff
@@ -223,4 +230,21 @@ private void removeEdge(final Edge<V> edge) {
         internalGraph.removeEdge(unpackEdge(edge));
     }
 
+    /**
+     * @author Rutger van den Berg
+     *         Compares two edges by their target vertex.
+     */
+    class EdgeComparatorByVertex implements Comparator<Edge<V>> {
```

Comment:
Lamda's were never meant to be used as syntactic sugar to create fancy one-liners, they have other advantages, for example its speed; they are way faster than creating class instances (see http://www.oracle.com/technetwork/java/jvmls2013kuksen-2014088.pdf). 

As an example (don't mind Github's bad indentation):

<code>
SortedSet<Edge<V>> output = new TreeSet<>((o1, o2) -> {
    int candidate = getDestination(o1).compareTo(getDestination(o2));
    return candidate == 0 ? getSource(o1).compareTo(getSource(o2)) : candidate;
});
</code>